src/NSubstitute/Extensions/ReturnsExtensions.cs

```diff
@@ -8,56 +8,41 @@ public static class ReturnsExtensions
         /// <summary>
         /// Set null as returned value for this call.
         /// </summary>
-        /// <typeparam name="T"></typeparam>
-        /// <param name="value"></param>
-        /// <returns></returns>
         public static ConfiguredCall ReturnsNull<T>(this T value) where T : class
         {
-            return value.Returns(i => null);
+            return value.Returns(default(T));
         }
 
         /// <summary>
         /// Set null as returned value for this call made with any arguments.
         /// </summary>
-        /// <typeparam name="T"></typeparam>
-        /// <param name="value"></param>
-        /// <returns></returns>
         public static ConfiguredCall ReturnsNullForAnyArgs<T>(this T value) where T : class
         {
-            return value.ReturnsForAnyArgs(i => null);
+            return value.ReturnsForAnyArgs(default(T));
         }
 
         /// <summary>
         /// Set null as returned value for this call.
         /// </summary>
-        /// <typeparam name="T"></typeparam>
-        /// <param name="value"></param>
-        /// <returns></returns>
         public static ConfiguredCall ReturnsNull<T>(this Task<T> value) where T : class
         {
-            return value.Returns(i => SubstituteExtensions.CompletedTask<T>(null));
+            return value.Returns(default(T));
         }
 
         /// <summary>
         /// Set null as returned value for this call.
         /// </summary>
-        /// <typeparam name="T"></typeparam>
-        /// <param name="value"></param>
-        /// <returns></returns>
         public static ConfiguredCall ReturnsNull<T>(this ValueTask<T> value) where T : class
         {
-            return value.Returns(i => SubstituteExtensions.CompletedValueTask<T>(null));
+            return value.Returns(default(T));
```

Comment:
I knew that this change will be a bit confusing, but we are _not_ returning `null` here instead of `Task` (which, of course, would be a mistake). 

The actual invocation chain is following:
`ReturnsNull<T>(this ValueTask<T> value)` =>
`Returns<T>(this ValueTask<T> value, T returnThis, params T[] returnThese)` =>
`ConfigureReturn<T>(MatchArgs matchArgs, T returnThis, T[] returnThese)`

So the `null` value we pass here is surrounded by `Task`/`ValueTask` in the next level extension method.

I love this change for the chaining, so we keep wrapping logic in a single place only.